### PR TITLE
Modifying configure file for enable to compile with private libraries

### PR DESCRIPTION
**Pull request by islamoglus**
Hi,
I need to compile Turn Server with private libraries ( Ex : BoringSSL )

This implementation provides to use private libraries for compilation. If user defined X_CFLAGS and X_LIBS variables for private library, "configure" file use this variables. Otherwise works as default.

Added options:
HIREDIS_CFLAGS HIREDIS_LIBS
PSQL_CFLAGS PSQL_LIBS
MYSQL_CFLAGS MYSQL_LIBS
MONGO_CFLAGS MONGO_LIBS
SSL_CFLAGS SSL_LIBS
EVENT_CFLAGS EVENT_LIBS

Thanks for reviewing.
Selahaddin.

### CODE REVIEW

**Comment by misi** [comment]
Thanks. I will check it.

File: configure [hunk 1]
```diff
@@ -1196,11 +1225,11 @@ ${ECHO_CMD} "# Generated by configure script #" >> Makefile
 ${ECHO_CMD} "#################################" >> Makefile
 ${ECHO_CMD} "ECHO_CMD = ${ECHO_CMD}" >> Makefile
 ${ECHO_CMD} "CC = ${CC}" >> Makefile
-${ECHO_CMD} "LDFLAGS += ${OSLIBS}" >> Makefile
-${ECHO_CMD} "DBLIBS += ${DBLIBS}" >> Makefile
-${ECHO_CMD} "CFLAGS += ${OSCFLAGS}" >> Makefile
+${ECHO_CMD} "LDFLAGS += ${P_LDFLAGS} ${OSLIBS}" >> Makefile
+${ECHO_CMD} "DBLIBS += ${P_DBLIBS} ${DBLIBS}" >> Makefile
+${ECHO_CMD} "CFLAGS += ${P_CFLAGS} ${OSCFLAGS}" >> Makefile
```
Comment (misi) [inline configure]: What is the reason to add P_CFLAGS in both on CFLAGS and also in DBCFLAGS too?

File: configure [hunk 2]
```diff
@@ -63,14 +63,19 @@ testlibevent2_comp() {
 }
 
 testhiredis() {
+    if [ -n "${HIREDIS_CFLAGS}" ] && [ -n "${HIREDIS_LIBS}" ]; then
+    	P_DBCFLAGS="${P_DBCFLAGS} ${HIREDIS_CFLAGS}"
+    	P_DBLIBS="${P_DBLIBS} ${HIREDIS_LIBS}"
+    	return 1
```
[the remark this answers is not in the document]
Comment (islamoglus) [reply configure]: thanks for point out, added

File: configure (hunk 1 above)
Comment (islamoglus) [reply configure]: removed

**Comment by islamoglus** [comment]
PR is re-designed for real compilation , Please check again.

Thanks
Selahaddin

**Pull request merged**